modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars.cc

```diff
@@ -42,21 +42,30 @@ namespace apollo {
 namespace perception {
 namespace lidar {
 
-const int PointPillars::kNumClass = 3;
-const int PointPillars::kMaxNumPillars = 12000;
-const int PointPillars::kMaxNumPointsPerPillar = 100;
+const float PointPillars::kPillarXSize = 0.25f;
+const float PointPillars::kPillarYSize = 0.25f;
+const float PointPillars::kPillarZSize = 8.0f;
+const float PointPillars::kMinXRange = -50.0f;
+const float PointPillars::kMinYRange = -50.0f;
+const float PointPillars::kMinZRange = -5.0f;
+const float PointPillars::kMaxXRange = 50.0f;
+const float PointPillars::kMaxYRange = 50.0f;
+const float PointPillars::kMaxZRange = 3.0f;
+const float PointPillars::kSensorHeight = 1.73f;
+// TODO(chenjiahao): kSensorHeight need to get from sensor's height param
+const int PointPillars::kNumClass = 10;
+const int PointPillars::kMaxNumPillars = 30000;
+const int PointPillars::kMaxNumPointsPerPillar = 60;
 const int PointPillars::kNumPointFeature = 4;
 const int PointPillars::kPfeOutputSize = kMaxNumPillars * 64;
-const int PointPillars::kGridXSize = 280;
-const int PointPillars::kGridYSize = 320;
-const int PointPillars::kGridZSize = 1;
+const int PointPillars::kGridXSize =
+    static_cast<int>((kMaxXRange - kMinXRange) / kPillarXSize);
+const int PointPillars::kGridYSize =
+    static_cast<int>((kMaxYRange - kMinYRange) / kPillarYSize);
+const int PointPillars::kGridZSize =
+    static_cast<int>((kMaxZRange - kMinZRange) / kPillarZSize);
 const int PointPillars::kRpnInputSize = 64 * kGridXSize * kGridYSize;
-const int PointPillars::kNumAnchorXInds = kGridXSize * 0.5;
-const int PointPillars::kNumAnchorYInds = kGridYSize * 0.5;
-const int PointPillars::kNumAnchorPerLoc = 6;
-// kNumAnchorPerLoc need to be equal to length of kAnchorRo
-const int PointPillars::kNumAnchor =
-    kNumAnchorXInds * kNumAnchorYInds * kNumAnchorPerLoc;
+const int PointPillars::kNumAnchor = 100 * 100 * 12 + 160 * 160 * 8;
```

Comment:
Regardng the const variables, if they don't change, you can consider static variable. If they may change with different settings, better use a config file to pass the parameters. @xiaoxq what's your take on this?

Reply:
Currently they have to be static const, cause memory malloced to cuda arrays at the build-time needs a fixed size provided by those build-time static variables. Is it possible to move these const variables into a config file while cuda arrays can be malloced memory at the build-time?

Reply:
Good question. Please try to group all parameters need by CUDA into a separate class with static const variables (e.g. class CudaParams { const static kVar1 ...... } and reference them in your PointPillars class so that we can **separate** parameters setting (with comments for each individual parameter) and workflow logic for a cleaner code organization purpose. What do you think?

Reply:
> Regardng the const variables, if they don't change, you can consider static variable. If they may change with different settings, better use a config file to pass the parameters. @xiaoxq what's your take on this?

Agree. Prefer `static constexpr`.

Reply:
how about we change the style in a separate commit ? I need some performance number this week.

Reply:
@jinghaomiao @weidezhang Parameters are now assigned values from a new class Params's static member variables and functions.